src/main/java/com/blibli/future/service/EmployeeTappingServiceImpl.java

```diff
@@ -167,22 +171,20 @@ public boolean addTapData(List<TapData> tapDataList) {
         }
     }
 
+    @Override
     public boolean addTapMachineFile(MultipartFile tapMachineFile) {
         List<TapData> tapDatas = new ArrayList<>();
-        List<String> inputData;
 
-        String fileName = tapMachineFile.getOriginalFilename();
+        List<String> inputData = fileReaderComponent.readFileAsStrings(tapMachineFile);
 
-        if(fileName.endsWith(".csv")) {
-            inputData = readCsvAsList(tapMachineFile);
-        } else {
-            return false; //Or should it throws some "Unsupported file Type" ?
+        if(inputData == null) {
```

Comment:
I mean if you throw an exception in here https://github.com/blibli-future/athens/blob/df563c877f90cf031a37ba0fc151121554490766/src/main/java/com/blibli/future/service/FileReaderComponentImpl.java#L32 you became "forced" to handle the special case.
```java 
try {
    List<String> inputData = fileReaderComponent.readFileAsStrings(tapMachineFile);
catch (UnsupportedFile | UnreadableFile e) {
    return false;
}
```